src/init.cpp

```diff
@@ -1889,6 +1889,7 @@ bool AppInitMain(const util::Ref& context, NodeContext& node)
     connOptions.nMaxConnections = nMaxConnections;
     connOptions.m_max_outbound_full_relay = std::min(MAX_OUTBOUND_FULL_RELAY_CONNECTIONS, connOptions.nMaxConnections);
     connOptions.m_max_outbound_block_relay = std::min(MAX_BLOCK_RELAY_ONLY_CONNECTIONS, connOptions.nMaxConnections-connOptions.m_max_outbound_full_relay);
+    connOptions.m_max_block_relay_anchor = std::min(MAX_BLOCK_RELAY_ONLY_ANCHORS, connOptions.m_max_outbound_block_relay);
```

Comment:
> Is there a reason this needs to be a `CConnMan::Options` option?

To be consistent with logic related to other connection limits, i.e., `MAX_OUTBOUND_FULL_RELAY_CONNECTIONS` and `MAX_BLOCK_RELAY_ONLY_CONNECTIONS`.

Reply:
I don't think we should copy bad patterns just because they already exist in the codebase. `Options` is used to pass parameters into `CConnman`'s initialize function, but `CConnman` already has everything it needs to calculate the max block relay anchor connections (the `MAX_BLOCK_RELAY_ONLY_ANCHORS` constant is defined in net, and `m_max_outbound_block_relay` is already passed in).

Adding this here is essentially placing logic for how to construct `CConnman` inside init.

Reply:
[Updated](https://github.com/bitcoin/bitcoin/pull/17428#issuecomment-688252761).